pandas/tests/frame/test_convert_to.py

```diff
@@ -274,6 +274,29 @@ def test_to_records_dtype(self, kwargs, expected):
             result = df.to_records(**kwargs)
             tm.assert_almost_equal(result, expected)
 
+    @pytest.mark.parametrize("kwargs,expected", [
```

Comment:
do we test with EA types / invalid types (misspelled for example)? These should just raise as numpy can't support them. (it probably does, but do we have a test for it)?

Reply:
Looks like both cases do actually raise, and I've added cases to cover them (former is a `ValueError`, latter is a `TypeError`)